v03_pipeline/lib/misc/validation.py

```diff
@@ -56,6 +56,30 @@ def validate_expected_contig_frequency(
             raise SeqrValidationError(msg)
 
 
+def validate_imputed_sex_ploidy(
+    mt: hl.MatrixTable,
+    sex_check_ht: hl.Table,
+) -> None:
+    mt = mt.select_cols(
+        discrepant=(
+            (
+                # All calls are diploid but the sex is Male/Unknown
+                hl.agg.all(mt.GT.is_diploid())
+                & (sex_check_ht[mt.s].predicted_sex != Sex.FEMALE.value)
+            )
+            | (
+                # At least one call is haploid but the sex is Female/Unknown
+                hl.agg.any(~mt.GT.is_diploid())
+                & (sex_check_ht[mt.s].predicted_sex != Sex.MALE.value)
+            )
+        ),
+    )
+    discrepant_rate = mt.aggregate_cols(hl.agg.fraction(mt.discrepant))
+    if discrepant_rate:
```

Comment:
I set this up on purpose so that it will throw if the `discrepant_rate` is greater than 0... so the most sensitive possible threshold.

Reply:
that tracks , but qq - wouldn't this just be checking if the discrepant_rate is none? or for float values, does 'if' check if it's non-zero?

Reply:
`0.0` is falsey!

Reply:
hmm that's cool what what I thought was up, but def wanted to check!